docs/source/docker/advanced/tutorial_al_with_docker.py

```diff
@@ -0,0 +1,144 @@
+"""
+
+.. _docker-active_learning:
+
+Documentation: Use the docker for active learning
+==============================================
+
+For running an active learning step with the Lightly docker, we need to perform
+3 steps:
+
+1. Create an embeddings.csv file with the Lightly docker.
+2. Add your active learning scores as an additonal column.
+3. Use the Lightly docker to perform a sampling on the scores.
+
+
+
+# Step 1:
+Run the docker as usual and as described in the getting started section.
+The only difference is that you set the number of samples to be sampled to 1,
+this prevents sampling.
+E.g. create and run a bash script with the following content:
+
+.. code::
+    INPUT_DIR=/path/to/your/dataset
+    SHARED_DIR=/path/to/shared
+    OUTPUT_DIR=/path/to/output
+
+    TOKEN=YOUR_TOKEN # put your token here
+    N_SAMPLES=1
+
+    docker run --ulimit nofile=32:32 --gpus all --rm -it \
+      -v ${INPUT_DIR}:/home/input_dir:ro  \
+      -v ${SHARED_DIR}:/home/shared_dir:ro \
+      -v ${OUTPUT_DIR}:/home/output_dir \
+      --ipc="host" --network="host" \
+      lightly/sampling:latest \
+      token=${TOKEN} \
+      lightly.loader.num_workers=4     \
+      stopping_condition.n_samples=${N_SAMPLES}\
+      method=coreset \
+      remove_exact_duplicates=True     \
+      enable_corruptness_check=True \
+      enable_training=True     \
+      lightly.trainer.max_epochs=100 \
+      dump_dataset=False \
+      upload_dataset=False
+
+After running it, this will create a embeddings.csv file
+in the output directory. Locate it and save the path to it.
+E.g. It may be found under
+`/path/to/output/2021-07-28/12:00:00/data/embeddings.cv`
+
+"""
+# %%
+# # Step 2:  Add active learning scores to the embeddings.cv file.
+# -----------------
+#
+# If you want to use your use predictions from your model as active learning scores,
+# you can use the Scorers from the lightly pip package.
+
+
+from typing import Iterable
+import csv
+import os
+
+"""
+Run your detection model here
+Use the scorers offered by lightly to generate active learning scores.
+"""
+
+# Let's assume that you have one active learning score for every image:
+# The order of the scores must match the order of filenames
+# in the embeddings.csv.
+scores: Iterable[float]
```

Comment:
More a personal opinion: Would not add typings. It's somehow blowing up the code and if the person going through this is not familiar with typings it might not help but more confuse.

Reply:
I added more comments to it.